syntax="proto3";

import "common.proto";
package MSG;


// chat
message ChatReq {
	string msg = 1;
	ChatType type = 2;
}

message ChatAns {
	ErrorCode err = 1;
}

message ChatNot {
	ChatData_ chats = 1;
	ChatType type = 2;
}


// create chat room
message CreateChatRoomReq {
	string name = 1;
	string greeting = 2;
}

message CreateChatRoomAns {
	ErrorCode err = 1;
	uint64 rid = 2;
	string rkey = 3;
}

// invite chat room
message InviteChatRoomReq {
	// room info
	uint64 rid = 1;
	string rkey = 2;

	// invite target info
	uint64 targetUid = 3;
	string targetName = 4;

}

message InviteChatRoomAns {
	ErrorCode err = 1;
}

message InviteChatRoomNot {
	// room info
	uint64 rid = 1;
	string rkey = 2;

	// room owner info
	uint64 ownerUid = 3;
	string ownerName = 4;

}

// enter chat room
message EnterChatRoomReq {
	// room info
	uint64 rid = 1;
	string rkey = 2;
}

message EnterChatRoomAns {
	ErrorCode err = 1;
}

message EnterChatRoomNot {
	uint64 uid = 1;
	string name = 2;
}


// leave chat room
message LeaveChatRoomReq {
	// room info
	uint64 rid = 1;
	string rkey = 2;
}

message LeaveChatRoomAns {
	ErrorCode err = 1;
}

message LeaveChatRoomNot {
	uint64 uid = 1;
	string name = 2;
}